src/vs/code/electron-main/menus.ts

```diff
@@ -196,6 +200,15 @@ export class VSCodeMenu {
 			updateMenu = true;
 		}
 
+		let newAutoHideMenuBar = config && config.window && config.window.autoHideMenuBar;
```

Comment:
Thanks for the review!

Everything makes sense, except I'm not quite sure why it should only update when the menu is hidden? For efficiency?